pkgs/development/compilers/swift/default.nix

```diff
@@ -0,0 +1,261 @@
+{ stdenv
+, cmake
+, coreutils
+, glibc
+, which
+, perl
+, libedit
+, ninja
+, pkgconfig
+, sqlite
+, swig
+, bash
+, libxml2
+, llvm
+, clang
+, python
+, ncurses
+, libuuid
+, libbsd
+, icu
+, autoconf
+, libtool
+, automake
+, libblocksruntime
+, curl
+, rsync
+, git
+, libgit2
+, binutils
+, fetchFromGitHub
+, paxctl
+, findutils
+#, systemtap
+}:
+
+let
+  v_major = "3.1";
+  version = "${v_major}-RELEASE";
+  version_friendly = "${v_major}";
+
+  tag = "refs/tags/swift-${version}";
+  fetch = { repo, sha256, fetchSubmodules ? false }:
+    fetchFromGitHub {
+      owner = "apple";
+      inherit repo sha256 fetchSubmodules;
+      rev = tag;
+      name = "${repo}-${version}-src";
+    };
+
+sources = {
+    clang = fetch {
+      repo = "swift-clang";
+      sha256 = "0820mx1ghfnk4p5595r1f313y1699jwi61zghfbwak5wgwgy7n4x";
+    };
+    llvm = fetch {
+      repo = "swift-llvm";
```

Comment:
Do you know if it really needs its own custom llvm or if it can use our pre-existing packages? I fear that we'll be missing patches and it'll reach out to all sorts of impure locations as a result.